README.md

```diff
@@ -229,6 +243,253 @@ Benchee.run %{"Awesome stuff" => fn -> magic end }
 your_teardown()
 ```
 
+When might this be useful?
```

Comment:
This is a really great addition!

Reply:
Yes, I really like this part! When would I want to use this? 👍